lib/cstring.h

```diff
@@ -103,6 +103,12 @@ class cstring {
         construct_from_shared(string.data(), string.length());
     }
 
+    // construct cstring from std::string_view. Do not use if possible, this is linear
```

Comment:
Our comment style really is all over the place. The one for this file is a lowercase double slashes one. One day I will set up doxygen to warn about inconsistencies.

Reply:
yes :(